iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/patent/OpenPatentWebServiceFacade.java

```diff
@@ -0,0 +1,327 @@
+    
+    public OpenPatentWebServiceFacade(int connectionTimeout, int readTimeout, 
+            String authHostName, int authPort, String authScheme, String authUriRoot,
+            String opsHostName, int opsPort, String opsScheme, String opsUriRoot, 
+            String consumerCredential, long throttleSleepTime, int maxRetriesCount) {
+        
+        this(buildHttpClient(connectionTimeout, readTimeout),
+                new HttpHost(authHostName, authPort, authScheme), authUriRoot, 
+                new HttpHost(opsHostName, opsPort, opsScheme), opsUriRoot,
+                consumerCredential, throttleSleepTime, maxRetriesCount, new JsonParser());
+        
+        // persisting for further serialization and deserialization
+        this.serDe = new SerDe(connectionTimeout, readTimeout, 
+                authHostName, authPort, authScheme, 
+                opsHostName, opsPort, opsScheme);
+    }
+    
+    /**
+     * Using this constructor simplifies object instantiation but also makes the whole object non-serializable.
+     */
+    protected OpenPatentWebServiceFacade(HttpClient httpClient, HttpHost authHost, String authUriRoot, HttpHost opsHost,
+            String opsUriRoot, String consumerCredential, long throttleSleepTime, int maxRetriesCount,
+            JsonParser jsonParser) {
+        this.httpClient = httpClient;
+        this.authHost = authHost;
+        this.authUriRoot = authUriRoot;
+        this.opsHost = opsHost;
+        this.opsUriRoot = opsUriRoot;
+        this.consumerCredential = consumerCredential;
+        this.throttleSleepTime = throttleSleepTime;
+        this.maxRetriesCount = maxRetriesCount;
+        this.jsonParser = jsonParser;
+    }
+
+    // ------------------- LOGIC----------------------------
+    
+    @Override
+    public String getPatentMetadata(ImportedPatent patent) throws Exception {
+        return getPatentMetadata(patent, getSecurityToken(), 0);
+    }
+
+    // ------------------- PRIVATE -------------------------
+    
+    private String getPatentMetadata(ImportedPatent patent, String securityToken, int retryCount) throws Exception {
+        
+        if (retryCount > maxRetriesCount) {
+            throw new PatentServiceException("number of maximum retries exceeded: " + maxRetriesCount);
+        }
+
+        HttpRequest httpRequest = buildPatentMetaRequest(patent, securityToken, opsUriRoot);
+        HttpResponse httpResponse = httpClient.execute(opsHost, httpRequest);
+
+        int statusCode = httpResponse.getStatusLine().getStatusCode();
+
+        switch (statusCode) {
+        case 200: {
+            HttpEntity entity = httpResponse.getEntity();
+            if (entity != null) {
+                return EntityUtils.toString(entity);
+            } else {
+                throw new PatentServiceException("got empty response, full status: " + httpResponse.getStatusLine()
+                        + ", server response: " + EntityUtils.toString(httpResponse.getEntity()));
```

Comment:
See my comment above: I should not be using `httpResponse.getEntity()` here because in this case it will be `null`. I have removed this from exception message.